ctapipe/coordinates/angular_frames.py

```diff
@@ -445,12 +197,122 @@ def telescope_to_camera(telescope_coord, camera_frame):
-    x_rotated = x_rotated.to(u.rad) * (focal_length / u.m)
-    y_rotated = y_rotated.to(u.rad) * (focal_length / u.m)
+    x_rotated = x_rotated * focal_length
+    y_rotated = y_rotated * focal_length
+
+    return CameraFrame(x_rotated * u.m, y_rotated * u.m, **telescope_coord.copy_properties())
+
+
+class AngularCoordinate(BaseCoordinate):
+    """
+
+    """
+    system_order = np.array(["CameraFrame", "TelescopeFrame",
+                             "NominalFrame", "HorizonFrame"])
+
+    transformations = np.array([camera_to_telescope, telescope_to_nominal,
+                                nominal_to_horizon])
+    reverse_transformations = np.array([telescope_to_camera, nominal_to_telescope,
+                                        horizon_to_nominal])
+
+    def __init__(self, focal_length=None, telescope_pointing=None, array_pointing=None,
+                 rotation=0 * u.deg):
+        """
+        Parameters
+        ----------
+        focal_length: ndarray
+            Focal length of telescope
+        telescope_pointing: HorizonFrame
+            Pointing direction of telescope
+        array_pointing: HorizonFrame
+            Pointing direction of array
+        rotation: ndarray
+            Rotation angle of camera in telescope
+        """
+        self.focal_length = focal_length
+        self.telescope_pointing = telescope_pointing
+        self.array_pointing = array_pointing
+        self.rotation = rotation
+
+        prop_dict = dict()
+        for key in self.__dict__:
+            prop_dict[key] = self.__dict__[key]
+        self.properties = prop_dict
+
+        return
+
+    def copy_properties(self):
+        """
+        Create a copy of the shared class parameters to share with other classes
+
+        Returns
+        -------
+        dict: Dictionary of shared class parameters
+        """
+        properties = self.properties
+        return properties
+
+
+class CameraFrame(AngularCoordinate, Cartesian2D):
+    """Camera coordinate frame.  The camera frame is a simple physical
+    cartesian frame, describing the 2 dimensional position of objects
+    in the focal plane of the telescope Most Typically this will be
+    used to describe the positions of the pixels in the focal plane
+    """
+
+    def __init__(self, x=None, y=None, **kwargs):
+        super().__init__(**kwargs)
+
+        self.x = x
+        self.y = y
+
+
+class TelescopeFrame(AngularCoordinate, Cartesian2D):
+    """Telescope coordinate frame.  Cartesian system to describe the
+    angular offset of a given position in reference to pointing
+    direction of a given telescope When pointing corrections become
+    available they should be applied to the transformation between
+    this frame and the camera frame
+    """
+
+    def __init__(self, x=None, y=None, **kwargs):
+        super().__init__(**kwargs)
+
+        self.x = x
+        self.y = y
+
+
+class NominalFrame(AngularCoordinate, Cartesian2D):
```

Comment:
Why is this a cartesian frame and not an AltAz like frame?

Reply:
If we do all of the reconstruction in 3D planes without projecting to a common system (as in HillasReconstructor), this frame is not actually needed, but it's still useful for displays, etc. @MaxNoe I think this is cartesian since it is a projection onto a common plane on the sky, not a sphere (so it's a spherical coordinate with a projection included).

Reply:
Thats right @kosack this frame is a projection of the angular offsets from the nominal array pointing direction so behaves like a cartesian system. BTW do we expect significant benefits from reconstructing in the spherical system? I do worry if we are doing some kind of fit based reconstruction you would waste a lot of time calculating angular offsets and it would just be easier to fit in an offset system like this.

Anyway I guess thats a philosophical question for another time.

Reply:
I think we can support both methods. The main advantage of doing it in 3D is less complexity, and no need to correct for telescopes pointing in different directions (divergent or slewing).

Reply:
Yeah your probably right, it feels more complicated to do it this way to me at the moment, but I just probably need to reorient my brain a bit 😀